experimental/list_formatter/src/provider.rs

```diff
@@ -6,20 +6,28 @@
 //!
 //! Read more about data providers: [`icu_provider`]
 
-use crate::options::{Type, Width};
+use crate::options::Width;
 use alloc::borrow::Cow;
-use core::ops::Index;
 use icu_provider::yoke::{self, *};
 
 pub mod key {
     //! Resource keys for [`list_formatter`](crate).
     use icu_provider::{resource_key, ResourceKey};
 
     // Resource key: symbols used for list formatting.
-    pub const LIST_FORMAT_V1: ResourceKey = resource_key!(ListFormatter, "list", 1);
+    pub const LIST_FORMAT_AND_V1: ResourceKey = resource_key!(ListFormatter, "list/and", 1);
+    pub const LIST_FORMAT_OR_V1: ResourceKey = resource_key!(ListFormatter, "list/or", 1);
+    pub const LIST_FORMAT_UNIT_V1: ResourceKey = resource_key!(ListFormatter, "list/unit", 1);
 }
 
 /// Symbols and metadata required for [`ListFormatter`](crate::ListFormatter).
+/// Absent values follow this fallback structure:
+/// ", " - start - middle
+///            |-- end - pair
+///            |     \ short_end - short_pair
+///            |               \ narrow_end - narrow_pair
+///             \ short_start - short_middle
+///                         \ narrow_start <- narrow_middle
```

Comment:
Another idea:

```
short_start
    start
    narrow_start
short_middle
    middle
    narrow_middle
short_pair
    pair
    narrow_pair
short_end
    end
    narrow_end
```

But actually I think we should store an `enum` which is either the pattern or a "flat" enum, and the "flat" enum references another field that should be used as fallback

Reply:
Will land this and work on a better fallback mechanism in a follow-up